src/EnergyPlus/ExtendedHI.cc

```diff
@@ -0,0 +1,499 @@
+                    state,
+                    tol,
+                    maxIter,
+                    SolFla,
+                    Ts,
+                    [&](Real64 Ts) { return (Ts - Ta) / Ra_un(Ts, Ta) + (Pc - Pa) / (Zs((Tc - Ts) / (Q - Qv(Ta, Pa))) + Za_un) - (Q - Qv(Ta, Pa)); },
+                    0.0,
+                    Tc);
+                Rs = (Tc - Ts) / (Q - Qv(Ta, Pa));
+                eqvar_name = static_cast<int>(EqvarName::Rs);
+                Real64 Ps = Pc - (Pc - Pa) * Zs(Rs) / (Zs(Rs) + Za_un);
+                if (Ps > phi_salt * pvstar(Ts)) {
+                    General::SolveRoot(
+                        state,
+                        tol,
+                        maxIter,
+                        SolFla,
+                        Ts,
+                        [&](Real64 Ts) { return (Ts - Ta) / Ra_un(Ts, Ta) + (phi_salt * pvstar(Ts) - Pa) / Za_un - (Q - Qv(Ta, Pa)); },
+                        0.0,
+                        Tc);
+                    Rs = (Tc - Ts) / (Q - Qv(Ta, Pa));
+                }
+                return {eqvar_name, Rs};
+            } else {
+                Rs = 0.0;
+                eqvar_name = static_cast<int>(EqvarName::DTcdt);
+                dTcdt = (1.0 / C) * flux3;
+                return {eqvar_name, dTcdt};
+            }
+        }
+    }
+
+    // Convert the find_T function
+    Real64 find_T(EnergyPlusData &state, int eqvar_name, Real64 eqvar)
+    {
+        Real64 T;
+        int SolFla;
+
+        if (eqvar_name == static_cast<int>(EqvarName::Phi)) {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return find_eqvar_phi(state, T, 1.0) - eqvar; }, 0.0, 240.0);
+        } else if (eqvar_name == static_cast<int>(EqvarName::Rf)) {
+            General::SolveRoot(
+                state,
+                tol,
+                maxIter,
+                SolFla,
+                T,
+                [&](Real64 T) { return (find_eqvar_Rf(state, T, std::min(1.0, Pa0 / pvstar(T)))) - eqvar; },
+                230.0,
+                300.0);
+        } else if (eqvar_name == static_cast<int>(EqvarName::Rs)) {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return find_eqvar_rs(state, T, Pa0 / pvstar(T)) - eqvar; }, 295.0, 350.0);
+        } else {
+            General::SolveRoot(
+                state, tol, maxIter, SolFla, T, [&](Real64 T) { return find_eqvar_dTcdt(state, T, Pa0 / pvstar(T)) - eqvar; }, 340.0, 1000.0);
+        }
+
+        return T;
+    }
+
+    Real64 heatindex(EnergyPlusData &state, Real64 Ta, Real64 RH)
+    {
+
+        // Ta: temperature in Kelvin
+        // RH: relative humidity in range of 0.0 to 1.0
+        // The function computes the extended heat index, in Kelvinn
+
+        auto const &HVACSystemRootSolver = state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver;
+        auto const HVACSystemRootSolverBackup = HVACSystemRootSolver;
```

Comment:
This is still making a copy of the root solver each call to heatIndex.  Which happens for each zone at each time step.  This is not OK.  I get the feeling you shouldn't use the HVACSystemRootSolver for this.  Instead just use SolveRoot?  

As an aside, I still think the find_eqvar with tuple and returning different things is still a bit too complex just to calculate heat index.  At this point I wouldn't hold it up just for that, but if it could be simplified further, that would be great.

Reply:
@Myoldmopar I just changed `find_eqvar` to two functions each returning a single value.

Reply:
btw I also just printed out the size of the things involved here. They're all 4 byte.
```
auto const &HVACSystemRootSolver = state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver;
auto const HVACSystemRootSolverBackup = HVACSystemRootSolver;
```
![image](https://github.com/user-attachments/assets/bb863447-27b9-4749-91b4-c7605112fec8)

Reply:
Removing the 
```
        auto const &HVACSystemRootSolver = state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver;
        auto const HVACSystemRootSolverBackup = HVACSystemRootSolver;
        state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver = HVACSystemRootSolverAlgorithm::Bisection;

...
        state.dataRootFinder->HVACSystemRootFinding.HVACSystemRootSolver = HVACSystemRootSolverBackup;
```
seems to cause some NaN in the unit test `HeatBalanceSurfaceManager_TestResilienceMetricReport`, when calculating this: `ExtendedHI::heatindex(state, 298.15, 0.5) `